src/controllers/apps/chat-app/message.controllers.js

```diff
@@ -154,5 +158,128 @@ const sendMessage = asyncHandler(async (req, res) => {
     .status(201)
     .json(new ApiResponse(201, receivedMessage, "Message saved successfully"));
 });
+const deleteMessage = asyncHandler(async (req, res) => {
+  //controller to delete chat messages and attachments
 
-export { getAllMessages, sendMessage };
+  const { chatId, messageId, attachmentId } = req.params;
+
+  //Find the chat based on chatId
+
+  const chat = await Chat.findOne({
+    _id: new mongoose.Types.ObjectId(chatId),
+  });
+
+  if (!chat) {
+    throw new ApiError(404, "Chat does not exist");
+  }
+  //Find the message based on message id
+
+  const message = await ChatMessage.findOne({
+    _id: new mongoose.Types.ObjectId(messageId),
+  });
+
+  if (!message) {
+    throw new ApiError(404, "Message does not exist");
+  }
+
+  // Check if user is the sender of the message
+  if (message.sender.toString() !== req.user._id.toString()) {
+    throw new ApiError(
+      401,
+      "You are not authorised to delete the message ,you are not the sender"
+    );
+  }
+
+  //Checking if 15 mins has passed since the message sent or not
+
+  const currentTime = new Date();
+  const messageCreatedAt = message.createdAt;
+  const timeDifferenceMinutes = (currentTime - messageCreatedAt) / (1000 * 60); //calculating the time difference in minutes
+
+  if (timeDifferenceMinutes > 15) {
+    throw new ApiError(
+      400,
+      "15 mins have passed you cannot delete this message"
+    );
+  }
+
+  // If frontend sends the attachment id and message has more than one attachment
+  //Then delete only the attachment sent by the user otherwise delete the whole message
+  if (attachmentId && message.attachments.length > 1) {
```

Comment:
No need to remove attachments through attachment id this controller must delete whole message document including attachments